src/c_arc.rs

```diff
@@ -42,11 +42,33 @@ pub struct CArc<T: ?Sized> {
 
     /// The same as [`Self::stable_ref`] but it never changes.
     #[cfg(debug_assertions)]
-    base_stable_ref: NonNull<T>,
+    base_stable_ref: StableRef<T>,
 
-    stable_ref: NonNull<T>,
+    stable_ref: StableRef<T>,
 }
 
+/// A stable reference, stored as a raw ptr.
+///
+/// It is stable (e.x. from a [`Box`] or a [`CBox`]),
+/// so it can be stored relative to its owner.
+/// Otherwise, it is essentially a `&T` (so immutable).
+#[derive(Debug)]
+struct StableRef<T: ?Sized>(NonNull<T>);
+
+impl<T: ?Sized> Clone for StableRef<T> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<T: ?Sized> Copy for StableRef<T> {}
+
+/// SAFETY: [`StableRef`]`<T>` is essentially `&T`, which is [`Send`] if `T: `[`Send`]`.
```

Comment:
This is only valid if the invariants on what is put into the ref is held, we should at least mention that here.